azurerm/internal/services/storage/resource_arm_hpc_cache_nfs_target.go

```diff
@@ -0,0 +1,288 @@
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Cache, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] HPC Cache NFS Target %q was not found (Resource Group %q, Cahe %q) - removing from state!", id.Name, id.ResourceGroup, id.Cache)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving HPC Cache NFS Target %q (Resource Group %q, Cahe %q): %+v", id.Name, id.ResourceGroup, id.Cache, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("cache_name", id.Cache)
+
+	if resp.StorageTargetProperties == nil {
+		return fmt.Errorf("Error retrieving HPC Cache NFS Target %q (Resource Group %q, Cahe %q): `properties` was nil", id.Name, id.ResourceGroup, id.Cache)
+	}
+	props := *resp.StorageTargetProperties
+
+	var hostName, usageModel string
+	if nfs3 := props.Nfs3; nfs3 != nil {
+		if nfs3.Target != nil {
+			hostName = *nfs3.Target
+		}
+		if nfs3.UsageModel != nil {
+			usageModel = *nfs3.UsageModel
+		}
+	}
+	d.Set("host_name", hostName)
+	d.Set("usage_model", usageModel)
+
+	if err := d.Set("namespace_junction", schema.NewSet(schema.HashResource(namespaceJunctionResource()), flattenNamespaceJunctions(props.Junctions))); err != nil {
+		return fmt.Errorf(`Error setting "namespace_junction" %q (Resource Group %q, Cahe %q): %w`, id.Name, id.ResourceGroup, id.Cache, err)
+	}
+
+	return nil
+}
+
+func resourceArmHPCCacheNFSTargetDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Storage.StorageTargetsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parsers.HPCCacheTargetID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, id.ResourceGroup, id.Cache, id.Name)
+	if err != nil {
+		return fmt.Errorf("Error deleting HPC Cache NFS Target %q (Resource Group %q, Cahe %q): %+v", id.Name, id.ResourceGroup, id.Cache, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for deletion of HPC Cache NFS Target %q (Resource Group %q, Cahe %q): %+v", id.Name, id.ResourceGroup, id.Cache, err)
+	}
+
+	return nil
+}
+
+func expandNamespaceJunctions(input []interface{}) *[]storagecache.NamespaceJunction {
+	result := make([]storagecache.NamespaceJunction, 0)
+
+	for _, v := range input {
+		b := v.(map[string]interface{})
+		result = append(result, storagecache.NamespaceJunction{
+			NamespacePath: utils.String(b["namespace_path"].(string)),
+			NfsExport:     utils.String(b["nfs_export"].(string)),
+			TargetPath:    utils.String(b["target_path"].(string)),
+		})
+	}
+
+	return &result
+}
+
+func flattenNamespaceJunctions(input *[]storagecache.NamespaceJunction) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	output := make([]interface{}, 0)
+
+	for _, v := range *input {
+		namespacePath := ""
```

Comment:
I think this ends up easier to follow:
```suggestion
		b := ap[string]interface{}{}
		if v := v.NamespacePath; v != nil {
			b["namespace_path"] = *v
		}
```
?

Reply:
Do you mean to directly set to the map in each `if` block, or just mean to introduce a auto var for easier dereference the property (e.g. `NamespacePath`).

If it's the former, then I am actually following the same way as it did (e.g.) in [linux_virtual_machine](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/compute/virtual_machine.go#L425), and I believe Tom suggest me or someone else to do it like that. The point is to always ensure a default value for that (though redundent in current SDK impl. but might useful if things changed).

If it's the latter, then I will fix that.